core/src/main/java/org/apache/carbondata/hadoop/CarbonInputSplit.java

```diff
@@ -359,7 +361,11 @@ public Segment getSegment() {
       this.length = in.readLong();
       this.version = ColumnarFormatVersion.valueOf(in.readShort());
       this.rowCount = in.readInt();
-      this.writeDeleteDelta = in.readBoolean();
+      int numberOfDeleteDeltaFiles = in.readInt();
+      deleteDeltaFiles = new String[numberOfDeleteDeltaFiles];
```

Comment:
Initialize deleteDeltaFiles, only if numberOfDeleteDeltaFiles != 0. Else Array size zero will be present and that's not a safe way of coding.

we cannot expect get method to check length of array, user may check just for null also.

Reply:
added null check to write/read delta files or not.